### PR TITLE
MAID-1235 nfs api refactor

### PR DESCRIPTION
**Pull request by hitman401**
NFS API Refactored. All tests are commented as it would fail

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/maidsafe_nfs/14)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @ustulation

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by ustulation** [comment]
<sup>**[src/directory_listing/directory_info.rs, line 38 [r3]](https://reviewable.io:443/reviews/maidsafe/maidsafe_nfs/14#-JvSaPNs3zYnQG4W-PAz)** ([raw file](https://github.com/maidsafe/maidsafe_nfs/blob/86ebd99d73681e8a08c331a33aee436caac38b9a/src/directory_listing/directory_info.rs#L38)):</sup>
why clone() ?
will not `(&self.id, self.typ_tag)` work?

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_nfs/14)

File: src/helper/directory_helper.rs [hunk 1]
```diff
@@ -29,177 +29,367 @@ impl DirectoryHelper {
...
 
     /// Return the DirectoryListing for the latest version
-    // TODO version parameter change it to value instead of &
-    pub fn get(&mut self, directory_id: &::routing::NameType) -> Result<::directory_listing::DirectoryListing, ::errors::NFSError> {
-        let structured_data = try!(::utility::get_structured_data(self.client.clone(), directory_id.clone(), ::VERSION_DIRECTORY_LISTING_TAG));
-        let versions = try!(::maidsafe_client::structured_data_operations::versioned::get_all_versions(&mut *self.client.lock().unwrap(), &structured_data));
-        let latest_version = versions.last().unwrap();
-        self.get_by_version(directory_id, &latest_version)
+    pub fn get(&self,
+               directory_key: (::routing::NameType, u64),
```
Comment (ustulation) [inline src/helper/directory_helper.rs]: try `(&nametype...)`

File: src/directory_listing/mod.rs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+    /// Get the unique ID that represents this DirectoryListing in the network
+    pub fn get_key(&self) ->  (::routing::NameType, u64) {
+        self.info.get_key()
+    }
+
+    /// Decrypts a directory listing
+    pub fn decrypt(client      : ::std::sync::Arc<::std::sync::Mutex<::maidsafe_client::client::Client>>,
+                   directory_id: &::routing::NameType,
+                   access_level: ::AccessLevel,
```
Comment (ustulation) [inline src/directory_listing/mod.rs]: try `&Accesslevel...` and update elsewhere

**Pull request merged**